test/e2e/network/topology_hints.go

```diff
@@ -124,12 +152,12 @@ var _ = common.SIGDescribe("Feature:Topology Hints", func() {
 			}
 		}
 
-		nodeList, err := c.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
-		framework.ExpectNoError(err)
 		nodesByZone := map[string]string{}
+		zonesWithNodes := map[string][]string{}
```

Comment:
Seems like we all we really need is one node name per zone, so a simpler data structure like `nodeForZone := map[zone]nodeName{}` would work. (Where zone and node name would just be strings).

Reply:
Good idea!